File: app/src/main/java/com/wojdor/hearthstonecards/data/service/interceptor/TokenInterceptor.java
package com.wojdor.hearthstonecards.data.service.interceptor;

import android.support.annotation.NonNull;

import com.wojdor.hearthstonecards.BuildConfig;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class TokenInterceptor implements Interceptor {

    private static final String API_KEY = "X-Mashape-Key";

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request request = chain.request().newBuilder()
                .addHeader(API_KEY, BuildConfig.HEARTHSTONE_API_KEY)
                .build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/data/service/interceptor/TokenInterceptor.kt
package com.wojdor.hearthstonecards.data.service.interceptor

import com.wojdor.hearthstonecards.BuildConfig
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class TokenInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request().newBuilder()
                .addHeader(API_KEY, BuildConfig.HEARTHSTONE_API_KEY)
                .build()
        return chain.proceed(request)
    }

    companion object {
        private const val API_KEY = "X-Mashape-Key"
    }
}
